docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,48 @@
+<!-- this document has been modified from the rOpenSci and dplyr CONTRIBUTING files --->
+<!-- https://github.com/ropensci/dotgithubfiles/blob/master/dotgithub/CONTRIBUTING.md -->
+<!-- https://github.com/tidyverse/dplyr/blob/master/.github/CONTRIBUTING.md -->
+
+# Contributing to hydromad
+
+This document outlines how to contribute changes to hydromad.
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before attemping to make a change. 
+
+Please note we have a code of conduct that we request you adhere to in all your interactions with this project (see below).
+
+### Prerequisites
+
+Before you make a substantial pull request, you should always file an issue and
+make sure someone from the team agrees that it’s a problem. If you’ve found a
+bug, create an associated issue using our `ISSUE_TEMPLATE.md` and provide steps on how to reproduce the bug.
+
+Prior to making a pull request, we highly recommend you refer to `PULL_REQUEST_TEMPLATE.md` for a full checklist of tasks that needs to be completed.
+
+### Pull request process
+
+The pull request process for contributing to hydromad is as follows:
+
+1. Create a branch in your Git fork and label it appropriately.
+2. Open a pull request to hydromad's `master` branch.
+3. Fill in the pull request template as required. The pull request template will automatically appear once the request has been made.
+4. Discuss any amendments your pull request requires with owners of the repository that need to be made before your pull request can be approved.
+
+#### Other notes
+
+*  We recommend that R code is in the tidyverse [style guide](http://style.tidyverse.org).
```

Comment:
Let's check the style recommendations fit hydromad before merging this

Reply:
I selected tidyverse because it's a fairly popular one as far as I'm aware. This decision is at your discretion :)

Reply:
Thanks - I appreciate the suggestion.
I suppose my question is: should we check what it does the existing codebase first, or should the decision just be on the basis of whether we agree with it?

Reply:
I've tried styler as an [add-in](https://www.tidyverse.org/blog/2017/12/styler-1.0.0/) and applied it to`R/armax.inverse.fit.R`. It reformatted the indentation and spacing of the script (e.g. changed DATA[,"Q"] to DATA[, "Q"]). I would recommend that you run styler on a couple of files yourself to make sure you're happy with the changes it proposes before proceeding.

I'd also like to add that tidyverse recommends snake_case for naming functions and objects. From what I can see, names in hydromad are a mix of camelCase and base R (i.e. where the words that make up a function name are separated by "." as in "armax.sim.R"). I think we may have discussed going over naming conventions at some point. If so, perhaps this needs to be brought forward if you want to proceed with the tidyverse style.

Reply:
The method.class names are necessary in some cases because we use S3 object-oriented methods, so the .sim naming I think was an extension of that.
I've opened an issue to deal with this separately from this PR: https://github.com/josephguillaume/hydromad/issues/52

```suggestion
*  We recommend that R code is in the tidyverse [style guide](http://style.tidyverse.org), with the exception of naming using snake_case. Naming conventions are being formalised - in the mean time, please use existing code as an example.
```